adapter/service/outlvlswitch/service.go

```diff
@@ -160,3 +174,37 @@ func (s *service) SetBinaryState(value bool) error {
 
 	return nil
 }
+
+// StartLevelTransition implements starting of the transition with validations and concurrent safety.
+func (s *service) StartLevelTransition(value string, startLvl int, duration time.Duration) error {
```

Comment:
Please add optional service properties (`sup_duration` and `sup_start_lvl`) to the service specification. Once done, please add some checks whether `startLvl` and `duration` should be passed to the controller or not, based on specification. Please reflect that in the tests as well.

Reply:
I have added it (12 lines below), but not sure I like that solution as I simply send default values to the controller. The other options I see is either additional interfaces (with different input parameters) or additional methods in the existing interface, but both of them seem to be over-engineering. 
Open for any ideas.